lib/ueberauth/strategy/cognito.ex

```diff
@@ -232,10 +246,22 @@ defmodule Ueberauth.Strategy.Cognito do
   end
 
   @doc """
-  Currently doesn't return any additional information.
+  Fetches the fields to populate the info section of the `Ueberauth.Auth` struct.
   """
-  def info(_conn) do
-    %Ueberauth.Auth.Info{}
+  def info(conn) do
+    id_token = conn.private[:cognito_id_token]
+    %Ueberauth.Auth.Info{
+      email:       id_token["email"],
+      name:        id_token["cognito:username"],
```

Comment:
`cognito:username` is already available at the top level as `auth.uid`. This should be mapped to `id_token["name"]` instead, per https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims

Reply:
cognito may not include `name` in the id_token if the authentication is done with SAML.
In this case, since `cognito:username` is always included, we adopted this one.

Reply:
I think `uid` should use `sub`, what do you think?
https://docs.aws.amazon.com/cognito/latest/developerguide/amazon-cognito-user-pools-using-tokens-with-identity-providers.html#amazon-cognito-user-pools-using-the-id-token


> Subject (sub)
> The sub claim is a unique identifier (UUID) for the authenticated user. It is not the same as the username which may not be unique.

Reply:
None of these are required, so that's okay: we're using this with SAML which is why we're using the `cognito:username` as the `uid`. Do you know how other OAuth2 implementations are handling the difference between `sub` and the username/e-mail?

Reply:
- ueberauth_google
	- https://github.com/ueberauth/ueberauth_google/blob/af3138ee63/lib/ueberauth/strategy/google.ex#L7
	- https://github.com/ueberauth/ueberauth_google/blob/af3138ee63/lib/ueberauth/strategy/google.ex#L67-L74
	- use `sub` by default
- ueberauth_facebook
	- https://github.com/ueberauth/ueberauth_facebook/blob/9af663399b/lib/ueberauth/strategy/facebook.ex#L8
	- https://github.com/ueberauth/ueberauth_facebook/blob/9af663399b/lib/ueberauth/strategy/facebook.ex#L80-L87
	- use `id`

https://docs.aws.amazon.com/cognito/latest/developerguide/amazon-cognito-user-pools-using-tokens-with-identity-providers.html#user-pool-id-token-payload
Because the sample URL above does not have `name` but `cognito:username`, I think it is better to treat `cognito:username` as `name` in cognito at least.

Reply:
The description of `name` from the Standard Claims is

>  End-User's full name in displayable form including all name parts, possibly including titles and suffixes, ordered according to the End-User's locale and preferences. 

That's not a good match for what `cognito:username` is.

It looks like you might be able to configure this as a part of the provider config, to override our default? https://github.com/ueberauth/ueberauth_github/blob/c9e6ed0ebc9f1de56f4b24d53903c375578fb56b/lib/ueberauth/strategy/github.ex#L53-L60

We'd accept a PR to make the UID configurable (if it isn't already), but the default for `uid` should remain `cognito:username`.